### PR TITLE
update gorm to reduce size of #1110

### PR DESCRIPTION
**Pull request by endophage**
@ecordell this is just to make your channels PR appear a little smaller. I want to get a feel for the amount of churn without the vendoring distorting it.

This just updates gorm and adds the associated inflection package. There are also a couple of necessary derefs as gorm's initialization now return a `*`.

Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

**Comment by cyli** [comment]
I think you also have to pull in the changes from the server and signer config tests, as per https://github.com/docker/notary/pull/1110/files#r105954902.

**Comment by endophage** [comment]
Ah thanks! I'm now getting the `killed` thing when just running tests on MacOS so I pushed to see what was failing :-P

File: server/storage/sql_models.go [hunk 1]
```diff
@@ -50,7 +50,7 @@ func (c Change) TableName() string {
 // CreateTUFTable creates the DB table for TUFFile
 func CreateTUFTable(db gorm.DB) error {
 	// TODO: gorm
-	query := db.Set("gorm:table_options", "ENGINE=InnoDB DEFAULT CHARSET=utf8").CreateTable(&TUFFile{})
+	query := db.AutoMigrate(&TUFFile{})
```
Comment (riyazdf) [inline server/storage/sql_models.go]: seems like we can still have the `Set()` call here and chain that into an `AutoMigrate`: http://jinzhu.me/gorm/database.html#migration

Comment (endophage) [inline server/storage/sql_models.go]: The switch to sqlite in a couple of tests mean the `ENGINE` options is no longer valid.

**Comment by cyli** [comment]
LGTM

**Comment by endophage** [comment]
Retrying the yubikey tests, it was another random connectivity failure (these are happening a lot recently). Will merge if they pass

**Comment by ecordell** [comment]
Thanks! I'll rebase the other PR on this.

**Comment by ecordell** [comment]
LGTM a little late :)

**Pull request merged**